packages/wallet/wdk/src/sequence/wallets.ts

```diff
@@ -461,6 +482,9 @@ function fromModulesTopology(topology: Config.Topology): Config.SapientSignerLea
     if (topology.address !== Constants.ZeroAddress) {
       throw new Error('signer-leaf-not-allowed-in-modules-topology')
     }
+  } else if (Config.isNestedLeaf(topology) && Config.isNode(topology.tree)) {
+    // This module is wrapped with a guard, it's always going to be the first element
+    modules = [...modules, ...fromModulesTopology(topology.tree[0])]
```

Comment:
Can we recursively call `fromModulesTopology` on a nestedLeaf instead? That way we cater to more nested arranged than just this guard set up

Reply:
I tried but the amount of work to support every flexible arrangement out there was a bit much. So I did the opposite 😄 We know what the shape of v3 wallet configs should be, we know what the sessions tree looks like (with and without guard), and we already assume a lot about the wallets anyway, so it was only natural to add those assumptions to modules.

If we need more flexibility, we can tackle it then